ui/components/app/assets/asset-list/network-filter/network-filter.tsx

```diff
@@ -0,0 +1,144 @@
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { setTokenNetworkFilter } from '../../../../../store/actions';
+import {
+  getCurrentChainId,
+  getCurrentNetwork,
+  getIsTestnet,
+  getPreferences,
+  getSelectedInternalAccount,
+  getShouldHideZeroBalanceTokens,
+  getNetworkConfigurationsByChainId,
+} from '../../../../../selectors';
+import { useI18nContext } from '../../../../../hooks/useI18nContext';
+import { SelectableListItem } from '../sort-control/sort-control';
+import { useAccountTotalFiatBalance } from '../../../../../hooks/useAccountTotalFiatBalance';
+import { Text } from '../../../../component-library/text/text';
+import {
+  Display,
+  JustifyContent,
+  TextColor,
+  TextVariant,
+} from '../../../../../helpers/constants/design-system';
+import { Box } from '../../../../component-library/box/box';
+import { AvatarNetwork } from '../../../../component-library';
+import UserPreferencedCurrencyDisplay from '../../../user-preferenced-currency-display';
+import { CHAIN_ID_TO_NETWORK_IMAGE_URL_MAP } from '../../../../../../shared/constants/network';
+
+type SortControlProps = {
+  handleClose: () => void;
+};
+
+const NetworkFilter = ({ handleClose }: SortControlProps) => {
+  const t = useI18nContext();
+  const dispatch = useDispatch();
+  const chainId = useSelector(getCurrentChainId);
+  const selectedAccount = useSelector(getSelectedInternalAccount);
+  const currentNetwork = useSelector(getCurrentNetwork);
+  const allNetworks = useSelector(getNetworkConfigurationsByChainId);
+  const isTestnet = useSelector(getIsTestnet);
+  const { tokenNetworkFilter, showNativeTokenAsMainBalance } =
+    useSelector(getPreferences);
```

Comment:
in the future , we can use this component somewhere else , like on this [PR](https://github.com/MetaMask/metamask-extension/pull/28180) for example  , my suggestion is to make `tokenNetworkFilter` and `setTokenNetworkFilter` as props in order to have something more generic.
check the changes i did on the PR i shared and if you want me to do , then i'll do it on my PR